web3-react/src/example/Example.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from 'react'
+import './Example.css'
+import { CurrentConfig } from '../config'
+import { ConnectionType, switchNetwork } from '../libs/connections'
+import { useWeb3React } from '@web3-react/core'
+import { ConnectionOptions } from '../libs/components/ConnectionOptions'
+import { CHAIN_INFO } from '../libs/constants'
+
+// Listen for new blocks and update the wallet
+const useOnBlockUpdated = (callback: (blockNumber: number) => void) => {
+  const { provider } = useWeb3React()
+  useEffect(() => {
+    if (!provider) {
+      return
+    }
+    const subscription = provider?.on('block', callback)
```

Comment:
If you do an early return, do you need a safe access with `?`? I think we could choose just one pattern?

Reply:
Good call - wondering if eslint could capture this too